formats/json-jackson/src/main/java/io/cloudevents/jackson/CloudEventSerializer.java

```diff
@@ -65,15 +65,23 @@ public CloudEventContextWriter withContextAttribute(String name, String value) t
         }
 
         @Override
-        public CloudEventContextWriter withContextAttribute(String name, Number value) throws CloudEventRWException {
+        public CloudEventContextWriter withContextAttribute(String name, Number value) throws CloudEventRWException
+        {
+            // Only Integer types are supported by the specification
+            if (value instanceof Integer) {
+                this.withContextAttribute(name, (Integer) value);
+            } else {
+                // Default to string representation for other numeric values
+                this.withContextAttribute(name, value.toString());
```

Comment:
Maybe we should fail here with a `CloudEventRWException`? Maybe this one? https://github.com/cloudevents/sdk-java/blob/master/api/src/main/java/io/cloudevents/rw/CloudEventRWException.java#L133

Reply:
Ideally we (or me!) would go through the entire SDK and remove any implementations of the deprecated function which would then result in it automagically becoming a String representation.

My worry would be if you changed this now to throw an exception it might break other stuff.

Reply:
> My worry would be if you changed this now to throw an exception it might break other stuff.

I don't think so. What is going to happen is that the sdk is becoming more strict when accepting jsons, which is fine. I prefer to increase the strictness of the input and then break the methods (which is going to happen in later sdk versions). Does it makes sense?

Reply:
Ack ... Will take a look